### ISSUE TITLE
callbacks added with rust are blocked by adblocker

### ISSUE THREAD

**Issue by jaroslaw-weber** [koute/stdweb#216]
I am querying button with id, then adding event in rust with "add_event_listener" function, but the function call is blocked by adblock.

```
btn.add_event_listener(move |_: ClickEvent| {
//do something
});
```
I could do a workaround for this, but it would be nice to have it out of the box.

**Comment by Pauan** [comment]
Is it also blocked if you add a similar callback in JS, like `btn.addEventListener(function () { ... }, true)`?

What is the exact error message that you get?

**Comment by jaroslaw-weber** [comment]
My script was not working with adblock, so I guessed that the listener was blocked, but actually downloading .wasm file was blocked. uBlock origin has blocking .wasm files on default :(

You can close the issue, the only solution would be whitelist the website.

**Comment by CryZe** [comment]
Yeah a lot of people have been running into that issue. Apparently people are writing lots of miners with wasm, so they are just straight up blocking it.

**Comment by koute** [comment]
Are they blocking the file based on the extension? If so then it would be pretty trivial to work around the issue.

**Comment by jaroslaw-weber** [comment]
Yes, it's about extension. I read somewhere that you don't need to use .wasm extension for wasm files, is it true?

**Comment by Pauan** [comment]
I'm not 100% sure, but my understanding is that it's based on the MIME type, not the extension. So yes, changing the extension should be fine (provided the MIME type is correct).

If changing the extension solves the problem, then those adblockers are being really naive and shortsighted: not only do they block legitimate programs, but the illegitimate programs can easily bypass the adblocker!

**Comment by CryZe** [comment]
That's the rule: `/.*(\/proxy|\.wasm|\.wsm|\.wa)$/$xmlhttprequest,domain=github.io|rawgit.com|reactor.cc|sickrage.ca|streamplay.to|tubetitties.com|vidfile.net`

**Comment by jaroslaw-weber** [comment]
Yes, changing the rule may work for me, but I would need to ask users to turn off the adblock or change the rule.

I guess workaround would be changing extension and wait until the wasm become so popular that adblock people need to reconsider how they handle wasm.
